src/formatDate.js

```diff
@@ -8,7 +8,47 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  // Extract the separator from the fromFormat and toFormat arrays
+  const fromSeparator = fromFormat.pop();
+  const toSeparator = toFormat.pop();
+
+  // Split the date using the fromFormat separator
+  const dateParts = date.split(fromSeparator);
+
+  // Create an object to map the format parts to their corresponding values
+  const dateMap = {};
+
+  fromFormat.forEach((part, index) => {
+    dateMap[part] = dateParts[index];
+  });
+
+  // Handle conversion from YY to YYYY and vice versa
+  const convertYear = (yearStr, fromType, toType) => {
+    if (fromType === 'YYYY' && toType === 'YY') {
+      return yearStr.slice(-2);
+    } else if (fromType === 'YY' && toType === 'YYYY') {
+      const year = parseInt(yearStr, 10);
+
+      return year < 30 ? `20${yearStr.padStart(2, '0')}` : `19${yearStr}`;
```

Comment:
There is a potential issue when converting from 'YY' to 'YYYY'. The `padStart` method is used on the original `yearStr`, which could already be a two-character string. If `yearStr` is '00' to '29', this will work as expected, but if `yearStr` is '30' or above, `padStart` will not add any padding, and the result will be '1930' to '1999' instead of the expected '1930' to '1929'. To fix this, ensure padding is not applied to `yearStr` since it's already in the correct format.